src/NuGet.Clients/NuGet.CommandLine/Common/SelfUpdater.cs

```diff
@@ -34,14 +35,22 @@ public class SelfUpdater
         });
 
         private readonly IConsole _console;
+        private readonly EmbeddedSignatureVerifier _embeddedSignatureVerifier;
 
-        public SelfUpdater(IConsole console)
+        public SelfUpdater(IConsole console, EmbeddedSignatureVerifier signatureVerifier)
```

Comment:
Seems low risk/low chance, but just wondering about the design in CLI world.... 
... we don't expect that anyone implements `SelfUpdater` and therefore this change breaks them, right?

In other words, should we have 2 constructors and then the Commands use the default ctor which itself invokes `EmbeddedSignatureVerifier.Create()`? And only when someone has a special signature verifier do they need to provide one.

Reply:
Thanks for the feedback. My understanding we care mostly about public API changes in NuGet.Core\* projects but not NuGet.Client\* projects.  I think this is the reason public API analyzer is not enabled for `NuGet.Clients\NuGet.CommandLine` project. @zivkan incase if he has a different perspective about this public API change. I did a quick search on GitHub to check if any repo is leveraging API in NuGet.CommandLine but didn't find any instance where this change may break.

Reply:
`NuGet.CommandLine` is the project that creates `nuget.exe`, which is a command line app, not a class library.